bugfix\permissions

christomaszewski opened the pull request:
Permissions request fixed to ask for both location and file write permissions at startup (needed for logging to work).

Other small changes:
Waypoint achieved threshold radius dropped to 3m. Try catches added in sensor data parsing code to catch number format exceptions when attempting to parse doubles from corrupted sensor data.

Commit:
Updated permissions request in main application. Changed Waypoint achievement threshold radius to 3m. Added try catch statements to handle corrupted sensor data.

psigen commented:
Looks pretty good, but we need to strip out all the changes to random buildfiles.  I can do this in a bit if you can't get to it.

I think we will probably just `.gitignore` those files in the future because they churn anytime the project is rebuilt.

psigen committed:
Removed unintended build system changes from pull request.

For future reference, here is how I undid the changes:

```
git diff --name-only feature/onetouch
git checkout feature/onetouch .idea/gradle.xml .idea/libraries/microlog4android_1_0_0.xml .idea/misc.xml
git checkout feature/onetouch .idea/libraries/support_*
git checkout feature/onetouch build.gradle gradle/wrapper/gradle-wrapper.properties server.iml
git commit .
```

psigen committed:
Fixed whitespace and wrapping.

psigen committed:
Added punctuation to a few comments.

psigen commented:
@christomaszewski for future reference, here is how I cleaned up the build file changes:

``` bash
git diff --name-only feature/onetouch
git checkout feature/onetouch \
        .idea/gradle.xml .idea/libraries/microlog4android_1_0_0.xml .idea/misc.xml \
        .idea/libraries/support_* \
        build.gradle gradle/wrapper/gradle-wrapper.properties server.iml
git commit .
```

psigen commented:
Ok this PR looks good now, just waiting for the build test to complete.
